core/models.py

```diff
@@ -14,13 +14,21 @@
 User = get_user_model()
 
 
+class GenericModelManager(models.Manager):

```

Comment:
Please change to ActiveOnlyManager or similar. 
This manager can be used on any model. And current name doesn't reflect what it does.